src/interfaces/vault.ts

```diff
@@ -304,6 +306,125 @@ export class VaultInterface<T extends ChainId> extends ServiceInterface<T> {
+   * @param vaultAddress
+   * @param tokenAddress
+   * @returns TokenAllowance
+   */
+  async getDepositAllowance(
+    accountAddress: Address,
+    vaultAddress: Address,
+    tokenAddress: Address
+  ): Promise<TokenAllowance> {
+    const spenderAddress = await this.getDepositSpenderAddress(vaultAddress, tokenAddress);
+    return this.yearn.tokens.allowance(accountAddress, tokenAddress, spenderAddress);
+  }
+
+  /**
+   * Fetch the token amount that has been allowed to be used for withdraw
+   * @param accountAddress
+   * @param vaultAddress
+   * @param tokenAddress
+   * @returns TokenAllowance
+   */
+  async getWithdrawAllowance(
+    accountAddress: Address,
+    vaultAddress: Address,
+    tokenAddress: Address
+  ): Promise<TokenAllowance> {
+    const spenderAddress = await this.getWithdrawSpenderAddress(vaultAddress, tokenAddress);
+    return this.yearn.tokens.allowance(accountAddress, vaultAddress, spenderAddress);
+  }
+
+  /**
+   * Approve the token amount to allow to be used for deposits
+   * @param accountAddress
+   * @param vaultAddress
+   * @param tokenAddress
+   * @param amount
+   * @param overrides
+   * @returns TransactionResponse
+   */
+  async approveDeposit(
+    accountAddress: Address,
+    vaultAddress: Address,
+    tokenAddress: Address,
+    amount?: Integer,
+    overrides?: CallOverrides
+  ): Promise<TransactionResponse> {
+    const spenderAddress = await this.getDepositSpenderAddress(vaultAddress, tokenAddress);
+    return this.yearn.tokens.approve(
+      accountAddress,
+      tokenAddress,
+      spenderAddress,
+      amount ?? MaxUint256.toString(),
+      overrides
+    );
+  }
+
+  /**
+   * Approve the token amount to allow to be used for withdraw
+   * @param accountAddress
+   * @param vaultAddress
+   * @param tokenAddress
+   * @param amount
+   * @param overrides
+   * @returns TransactionResponse
+   */
+  async approveWithdraw(
+    accountAddress: Address,
+    vaultAddress: Address,
+    tokenAddress: Address,
+    amount?: Integer,
+    overrides?: CallOverrides
+  ): Promise<TransactionResponse> {
+    const spenderAddress = await this.getWithdrawSpenderAddress(vaultAddress, tokenAddress);
+    return this.yearn.tokens.approve(
+      accountAddress,
+      vaultAddress,
+      spenderAddress,
+      amount ?? MaxUint256.toString(),
+      overrides
+    );
+  }
+
+  private async getDepositSpenderAddress(vaultAddress: Address, tokenAddress: Address): Promise<Address> {
+    if (isNativeToken(tokenAddress)) return vaultAddress;
+
+    const willDepositUnderlyingToken = await this.isUnderlyingToken(vaultAddress, tokenAddress);
+    const shouldUsePartnerService = this.shouldUsePartnerService(vaultAddress);
+
+    if (willDepositUnderlyingToken && shouldUsePartnerService) {
+      const partnerTrackingContractAddress = await this.yearn.services.partner?.address;
+      if (!partnerTrackingContractAddress) throw new SdkError("Partner Tracking Contract Address not defined");
```

Comment:
There is no actualy need for this, shouldUsePartnerService will always make sure that partner is defined

Reply:
Its actually needed as a typeguard, because if not, we get a compiler error returning partnerTrackingContractAddress since typescript does not infer that logic.

Reply:
you can do this.yearn.services.partner!.address

But I got no strong feelings about this one, so up to you

Reply:
Since we have non-null assertions forbidden in eslint rules, this implementation is the way to go